### PR TITLE
Implement Edit Alias Command

### PR DESCRIPTION
**Pull request by aleksiuno**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR introduces the Edit Alias command required for Search Alias grid > Add alias.
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Manual testing is not required. Automatic Unit/Behat tests are implemented and the command is not yet used anywhere except the tests.
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/32667
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

### CODE REVIEW

File: src/Adapter/Alias/Repository/AliasRepository.php [hunk 1]
```diff
@@ -164,4 +168,43 @@ public function partialUpdate(Alias $alias, array $propertiesToUpdate, string $e
...
+            ->setParameter('searchTerm', $searchTerm)
+            ->execute()
+            ->fetchFirstColumn()
+        ;
+
+        if (empty($alisIds)) {
+            return;
+        }
+
+        foreach ($alisIds as $currentAliasId) {
```
Comment (zuk3975) [inline src/Adapter/Alias/Repository/AliasRepository.php]:
```suggestion
        foreach ($aliasIds as $currentAliasId) {
```

File: src/Adapter/Alias/Repository/AliasRepository.php [hunk 2]
```diff
@@ -164,4 +168,43 @@ public function partialUpdate(Alias $alias, array $propertiesToUpdate, string $e
...
+    /**
+     * Deletes all related aliases
+     *
+     * @param string $searchTerm
+     */
+    public function deleteAliasesBySearchTerm(string $searchTerm): void
+    {
+        $exceptions = [];
+
+        $alisIds = $this->connection->createQueryBuilder()
```
Comment (zuk3975) [inline src/Adapter/Alias/Repository/AliasRepository.php]:
```suggestion
        $aliasIds = $this->connection->createQueryBuilder()
```

File: src/Adapter/Alias/Repository/AliasRepository.php [hunk 3]
```diff
@@ -164,4 +168,43 @@ public function partialUpdate(Alias $alias, array $propertiesToUpdate, string $e
...
+            ->addSelect('a.id_alias')
+            ->from($this->dbPrefix . 'alias', 'a')
+            ->where('a.search = :searchTerm')
+            ->setParameter('searchTerm', $searchTerm)
+            ->execute()
+            ->fetchFirstColumn()
+        ;
+
+        if (empty($alisIds)) {
```
Comment (zuk3975) [inline src/Adapter/Alias/Repository/AliasRepository.php]:
```suggestion
        if (empty($aliasIds)) {
```

File: src/Core/Domain/Alias/Exception/AliasExistsException.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Alias\Exception;
+
+class AliasExistsException extends AliasException
```
Comment (zuk3975) [inline src/Core/Domain/Alias/Exception/AliasExistsException.php]: I don't see any usages of this class

**Review by zuk3975 — CHANGES_REQUESTED** [review]
unused exception class should be removed and there is a typo in one var

**Review by zuk3975 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Comment by matthieu-rolland** [comment]
thank you @aleksiuno :+1:

**Pull request merged**